pyxrf/model/scan_metadata.py

```diff
@@ -0,0 +1,213 @@
+        return key in self._values
+
+    def __iter__(self):
+        return self._values.__iter__()
+
+    def keys(self):
+        return self._values.keys()
+
+    def values(self):
+        return self._values.values()
+
+    def items(self):
+        return self._values.items()
+
+    def get_metadata_dictionary(self):
+        """
+        Returns reference to metadata dictionary ``self._values``
+        """
+        return self._values
+
+    def is_metadata_available(self):
+        """
+        Returns True if at least one metadata key:value pair is available.
+        Otherwise returns False.
+        """
+        return bool(self._values)
+
+    def get_formatted_output(self):
+        """
+        Returns formatted metadata in the form ready for printing.
+        Formatting is performed based on specifications located in ``self._print_order``
+        list. Key names are replaced by descriptions from ``self.descriptions`` if
+        the available.
+        """
+        str_out = ""
+
+        printed_keys = set()
+        flag_empty_line = False  # Indicates if empty line was just printed
+
+        for ptn in self._gen_default_print_order():
+            # We don't want to print multiple empty lines in a row
+            if ptn == "" and not flag_empty_line:
+                str_out += "\n"
+                flag_empty_line = True
+                continue
+
+            for key, v in self._values.items():
+                if re.search(f"^{ptn}$", key) and (key not in printed_keys):
+                    flag_empty_line = False  # Something is getting printed
+                    # Mark the entry as printed
+                    printed_keys.add(key)
+                    # Extracted printable expression for the key
+                    s_key = key
+                    if key in self.descriptions and self.descriptions[key]:
+                        s_key = self.descriptions[key]
+                        # Capitalize the first letter
+                        s_key = s_key[:1].upper() + s_key[1:]
+
+                    val = v
+
+                    # If the value is a string and it is too long, then add line breaks
+                    #   so that it fits the width of a normal sized window
+                    if isinstance(val, str):
+                        # Now check if 'v' is too long and insert line breaks if needed
+                        n_max = 60  # Maximum number of symbols in the line
+                        n_gap = len(s_key) + 2  # Size of the left margin
+                        s_insert = "\n" + " " * n_gap  # New line and spaces to form a paragraph
+
+                        nc = n_max
+                        while True:
+                            n = val.find(' ', nc)
+                            if n < 0:  # n is -1 if nc > len(val).
+                                break  # Exit here if the line is too short as well
+                            val = val[:n] + s_insert + val[n+1:]
```

Comment:
I don't think it is correct. It will add an extra space to the output, which will not be noticeable in this particular case, but in general it is incorrect. The space at location ``n`` should be deleted.

Reply:
Oh, I see what you are doing. I didn't realize it was trimming the blank spaces.